Add kwinputs to Kineto Traces

sraikund16 opened the pull request:
Summary: On the autograd side of things, we are currently saving the kwinputs but we aren't doing anything with them on the profiler side. This diff enables the use of the kwinputs for both FunctionEvents and Chrome Traces.

Test Plan: Added unit testing for both chrome traces and FunctionEvents. Used RecordFunctionFast to test kwinputs since test already had kwargs being passed in but not tested.

Differential Revision: D59472345

davidberard98 commented on torch/csrc/profiler/collection.h:
```diff
@@ -591,6 +595,9 @@ class TORCH_API ThreadLocalSubqueue {
     // report extra metadata, i.e. collective communication meta
     AppendOnlyList<extra_meta_t, BlockSize> extra_meta_;
 
+    // report wargs
```
typo

davidberard98 commented on torch/csrc/profiler/util.cpp:
```diff
@@ -292,6 +292,18 @@ std::string strListToStr(const std::vector<std::string>& types) {
+    std::stringstream ss;
+    ss.str("");
+    ss << "\"";
+    ss << val;
```
do we need to sanitize this value?

davidberard98 replied:
ok sorry, changed my mind haha - probably fine as it was before; I think this probably gets sanitized on the kineto side?

if you don't mind, would you be able to sanity check whatever choice you make with a test? I'm just slightly over-concerned about this because I think we've had SEVs before from broken traces due to unsanitized text in the json

sraikund16 replied:
I think kineto just sanitizes for backslashes and new lines. We might wanna keep this as is and then later on try to unify all the sanitizations that happen in different areas in the code with a follow up.

sraikund16 replied:
Will add a test still

davidberard98 reviewed (commented):
mostly lgtm, maybe just check if we need to sanitize any of the inputs?

albanD commented on torch/csrc/autograd/init.cpp:
```diff
@@ -228,6 +228,15 @@ PyObject* THPAutograd_initExtension(PyObject* _unused, PyObject* unused) {
             return as_pyobj;
           })
+      .def(
+          "kwinputs",
```
Should we use the profiler/python/init.cpp to bind these profiler APIs?

davidberard98 replied:
good idea, maybe we can do this in a separate PR though since it would involve moving all the KinetoEvent methods?

sraikund16 replied:
Agreed, can follow up

sraikund16 committed:
Add kwinputs to Kineto Traces (#130373)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/130373

On the autograd side of things, we are currently saving the kwinputs but we aren't doing anything with them on the profiler side. This diff enables the use of the kwinputs for both FunctionEvents and Chrome Traces.

Test Plan: Added unit testing for both chrome traces and FunctionEvents. Used RecordFunctionFast to test kwinputs since test already had kwargs being passed in but not tested.

Reviewed By: davidberard98

Differential Revision: D59472345
